Assets/Scripts/Features/Gameplay/Camera/CameraZooming.cs

```diff
@@ -0,0 +1,88 @@
+﻿using CM.Events;
+using UnityEngine;
+
+/// <summary>
+/// A class for zooming the Camera in and out.
+/// This script needs to be placed on the camera object.
+/// </summary>
+public class CameraZooming : MonoBehaviour
+{
+    /// <summary>
+    /// A simple event to call when the camera is zooming in or out.
+    /// </summary>
+    public event SimpleEvent OnZooming;
+
+    private float _minZoomDistanceZ;
+    private float _maxZoomDistanceZ;
+    private float _minZoomDistanceY;
+    private float _maxZoomDistanceY;
+
+    private bool _isZooming = false;
+
+    [SerializeField]
+    private float _zoomSpeedMultiplier = 0.5f;
+    [SerializeField]
+    private float _minZoomDistance = 0.0f;
+    [SerializeField]
+    private float _maxZoomDistance = 5.0f;
+
+    [SerializeField]
+    private bool _canZoom = false;
+
+    private void Start()
+    {
+        // Makes the min and max zoom level relative to the original camera position.
+        _minZoomDistanceZ = Mathf.Cos(transform.rotation.x) * _minZoomDistance + transform.position.z;
+        _maxZoomDistanceZ = Mathf.Cos(transform.rotation.x) * _maxZoomDistance + transform.position.z;
+
+        _minZoomDistanceY = Mathf.Sin(transform.rotation.x) * _minZoomDistance + transform.position.y;
+        _maxZoomDistanceY = Mathf.Sin(transform.rotation.x) * _maxZoomDistance + transform.position.y;
+    }
+
+    private void Update()
+    {
+        if (_canZoom)
+        {
+
```

Comment:
Everything inside this Update loop is dependent on this single if-statement.
For better practices make this undependent like below:
`
if (!_canZoom)
    return;

other code;
`